integration-tests/src/main/java/org/apache/druid/testing/utils/AbstractTestQueryHelper.java

```diff
@@ -146,24 +147,29 @@ private void testQueries(String url, List<QueryResultType> queries) throws Excep
     for (QueryResultType queryWithResult : queries) {
       LOG.info("Running Query %s", queryWithResult.getQuery());
       List<Map<String, Object>> result = queryClient.query(url, queryWithResult.getQuery());
-      if (!QueryResultVerifier.compareResults(result, queryWithResult.getExpectedResults(),
-                                              queryWithResult.getFieldsToTest()
-      )) {
+      Optional<String> resultsComparison = QueryResultVerifier.compareResults(result, queryWithResult.getExpectedResults(),
+                                                                                             queryWithResult.getFieldsToTest());
+      if (resultsComparison.isPresent()) {
         LOG.error(
             "Failed while executing query %s \n expectedResults: %s \n actualResults : %s",
             queryWithResult.getQuery(),
             jsonMapper.writeValueAsString(queryWithResult.getExpectedResults()),
             jsonMapper.writeValueAsString(result)
         );
-        failed = true;
+        Assert.fail(StringUtils.format(
+            "Results mismatch while executing the query %s.\n"
+            + "Expected results: %s\n"
+            + "Actual results: %s\n"
+            + "Mismatch error: %s\n",
+            queryWithResult.getQuery(),
+            jsonMapper.writeValueAsString(queryWithResult.getExpectedResults()),
+            jsonMapper.writeValueAsString(result),
+            resultsComparison.get()
+        ));
```

Comment:
Removed the expected and actual results part because that would be the two large JSON blobs that you mentioned here. The row number should be informed to the user in the test case. As far as the formatting of the results is concerned, I think it might be difficult to do so once the file has been deserialized into an object. One way would be to serialize the expected result the opposite way the deserialization has happened.